app/src/main/java/com/keylesspalace/tusky/components/preference/AccountPreferencesFragment.kt

```diff
@@ -253,7 +250,8 @@ class AccountPreferencesFragment : PreferenceFragmentCompat(), Injectable {
                     key = PrefKeys.ALWAYS_SHOW_SENSITIVE_MEDIA
                     setTitle(R.string.pref_title_alway_show_sensitive_media)
                     isSingleLineTitle = false
-                    isChecked = accountManager.activeAccount?.alwaysShowSensitiveMedia ?: false
+                    val x = accountManager.activeAccount?.alwaysShowSensitiveMedia ?: false
+                    preferenceDataStore = AccountPreferenceDataStore(accountManager.activeAccount!!)
                     setOnPreferenceChangeListener { _, newValue ->
                         updateAccount { it.alwaysShowSensitiveMedia = newValue as Boolean }
```

Comment:
Updating the account object (and maybe saving it) could also move to that DataStore class.